### PR TITLE
Apply 'types' field in tsconfigs

### PR DESCRIPTION
**Pull request by snario**

### CODE REVIEW

File: packages/cf.js/tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
   "compilerOptions": {
     "sourceRoot": ".",
     "declarationDir": "dist/",
-    "outDir": "dist/"
+    "outDir": "dist/",
+    "types": ["jest"]
```
Comment (cf19drofxots) [inline packages/cf.js/tsconfig.json]: Why is this needed?

Comment (snario) [reply packages/cf.js/tsconfig.json]: So we don't need to import `expect`, `beforeAll`, etc and also so we don't need to rely on a global in typescript-typings

Comment (cf19drofxots) [reply packages/cf.js/tsconfig.json]: We import those? Where?

Comment (snario) [reply packages/cf.js/tsconfig.json]: It is kind of subtle at the moment. `@types/jest` happens to get hoisted and in the root tsconfig we have a typeRoots config option that fetches it. However if we explicitly declare this on a per-package level it is safer and does not assume hoisting. Super tiny detail
